packages/xo-server/src/_MultiKeyMap.js

```diff
@@ -1,58 +1,88 @@
-const kValue = Symbol('value')
+class Node {
+  constructor(value) {
+    this.children = new Map()
+    this.value = value
+  }
+}
 
-function del(map, i, n, keys) {
+function del(node, i, n, keys) {
   if (i === n) {
-    return map.delete(kValue)
+    if (node instanceof Node) {
+      node.value = undefined
+      return node
+    }
+    return
+  }
+  if (!(node instanceof Node)) {
+    return node
   }
   const key = keys[i]
-  const child = map.get(key)
+  const { children } = node
+  const child = children.get(key)
   if (child === undefined) {
-    return false
+    return node
   }
-  const deleted = del(child, i + 1, n, keys)
-  if (child.size === 0) {
-    map.delete(key)
+  const newChild = del(child, i + 1, n, keys)
+  if (newChild === undefined) {
+    if (children.size === 1) {
+      return node.value
+    }
+    children.delete(key)
+  } else if (newChild !== child) {
+    children.set(key, newChild)
   }
-  return deleted
+  return node
 }
 
-function get(map, i, n, keys) {
-  if (i === n) {
-    return map.get(kValue)
-  }
-  const child = map.get(keys[i])
-  if (child !== undefined) {
-    return get(child, i + 1, n, keys)
-  }
+function get(node, i, n, keys) {
+  return node instanceof Node
+    ? i === n
```

Comment:
I'd reverse the tests (`i === n` then `node instanceof Node`) to keep it consistent with `del` and `set`.